parity/cli/mod.rs

```diff
@@ -562,6 +562,15 @@ usage! {
 			"--ipfs-api-cors=[URL]",
 			"Specify CORS header for IPFS API responses. Special options: \"all\", \"none\".",
 
+		["Light Client Options"]
+			ARG arg_on_demand_nb_retry: (Option<usize>) = None, or |c: &Config| c.light.as_ref()?.on_demand_nb_retry,
+			"--on_demand_nb_retry=[RETRIES]",
+			"Specify maximum number of retry query to send to other node for a query.",
+
+			ARG arg_on_demand_inactive_time_limit: (Option<u64>) = None, or |c: &Config| c.light.as_ref()?.on_demand_inactive_time_limit,
+			"--on_demand_inactive_time_limit=[MS]",
+			"Specify maximum total time limit for a light client query to stay inactive. O for no limit.",
```

Comment:
It is currently in MilliSeconds ([MS]), and yes 0 is no limit wait forever; it is not that unsafe because it is combined with the maximum number of retries (at first I implement it without any timeout).

Reply:
Here it is not a hard timelimit, it is really an inactive time (if new peers are added regularly the request could run for way longer than this value). Inactive means that there was no new query for some time, if I got a new query in between the inactive time is reset to 0.